core/src/main/java/org/apache/iceberg/rest/requests/PlanTableScanRequest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import java.util.List;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+
+public class PlanTableScanRequest implements RESTRequest {
+  private Long snapshotId;
+  private List<String> select;
+  private Expression filter;
+  private Boolean caseSensitive;
+  private Boolean useSnapshotSchema;
+  private Long startSnapshotId;
+  private Long endSnapshotId;
+  private List<String> statsFields;
+
+  public Long snapshotId() {
+    return snapshotId;
+  }
+
+  public List<String> select() {
+    return select;
+  }
+
+  public Expression filter() {
+    return filter;
+  }
+
+  public Boolean caseSensitive() {
+    return caseSensitive;
+  }
+
+  public Boolean useSnapshotSchema() {
+    return useSnapshotSchema;
+  }
```

Comment:
I think these should always be `boolean`. From a model perspective I think you'll want these methods to return some value especially since we have defaults defined. This will also simplify the parser logic where you don't have to if/else these cases

Reply:
Will fix this.